gui/src/main/java/DiceGameGUI.java

```diff
@@ -300,6 +300,18 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
     }// </editor-fold>//GEN-END:initComponents
 
     private void btnResetActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_btnResetActionPerformed
+        //Display pop up showing score before starting new game.
+        JOptionPane.showMessageDialog(
+            null, 
+            "Game Over! You save your score of " 
+            + thisGame.getCurrentTotal()
+            + "\nFinal Score: "
+            + thisGame.getScore(),
+            "Game Over!",
+            JOptionPane.OK_OPTION);

```

Comment:
This need to be in an if(thisGame.getCurrentTotal() <23) so there's no redundant message box after the hit or go over 23